src/invoice/printable-invoice/printable-invoice.component.tsx

```diff
@@ -114,13 +177,42 @@ const PrintableInvoice: React.FC<PrintableInvoiceProps> = ({ bill, patient, comp
               )}
             </DataTable>
           </div>
-
-          <div className={styles.totalContainer}>
-            {Object.entries(invoiceTotal).map(([key, val]) => (
-              <p key={key} className={styles.itemTotal}>
-                <span className={styles.itemHeading}>{key}</span>: <span className={styles.itemLabel}>{val}</span>
-              </p>
-            ))}
+          {bill?.payments?.length > 0 && (
+            <div className={styles.paymentHistoryContainer}>
+              <DataTable rows={paymentHistoryRows} headers={paymentHistoryHeaders} size={responsiveSize}>
+                {({ rows, headers, getRowProps, getTableProps }) => (
+                  <TableContainer>
+                    <Table {...getTableProps()} aria-label={t('paymentHistory', 'Payment History')}>
+                      <TableHead>
+                        <TableRow>
+                          {headers.map((header) => (
+                            <TableHeader key={header.key}>{header.header}</TableHeader>
+                          ))}
+                        </TableRow>
+                      </TableHead>
+                      <TableBody>
+                        {rows.map((row) => (
+                          <TableRow key={row.id} {...getRowProps({ row })}>
+                            {row.cells.map((cell) => (
+                              <TableCell key={cell.id}>{cell.value}</TableCell>
+                            ))}
+                          </TableRow>
+                        ))}
+                      </TableBody>
+                    </Table>
+                  </TableContainer>
+                )}
+              </DataTable>
+            </div>
+          )}
+          <div className={styles.balanceContainer}>
+            <span className={styles.itemHeading}>{t('balance', 'Balance')}:</span>{' '}
```

Comment:
```suggestion
            <span className={styles.itemHeading}>{t('balance', 'Balance')}:</span>
```
Is this empty space necessary?